pyiceberg/catalog/rest.py

```diff
@@ -289,12 +291,25 @@ def auth_url(self) -> str:
         else:
             return self.url(Endpoints.get_token, prefixed=False)
 
+    def _extract_optional_oauth_params(self) -> Dict[str, str]:
+        set_of_optional_params = {AUDIENCE, RESOURCE}
```

Comment:
Can we include `SCOPE` as well since it is optional? This is pending #484.

Reply:
Yes, we can but scope has default value and others ones do not have in iceberg Java. Also to keep parity with Java version I have kept scope separate. I was thinking once we consolidate these params in Java, we can then do it here. WDYT?

Reply:
Can we extend this method like this? So that all optional parameters are handled by a single method. I feel this is a bit more organized. Not a blocker though.
```
def _extract_optional_oauth_params(self) -> Dict[str, str]:
        optional_oauth_param = {}
        optional_oauth_param[SCOPE] = self.properties.get(SCOPE) or CATALOG_SCOPE

        set_of_optional_params = {AUDIENCE, RESOURCE}
        for param in set_of_optional_params:
            if param_value := self.properties.get(param):
                optional_oauth_param[param] = param_value

        return optional_oauth_param
```

Reply:
I also like the suggestion, I think we can simplify it a bit more:

```python
def _extract_optional_oauth_params(self) -> Dict[str, str]:
    optional_oauth_param = {
        SCOPE: self.properties.get(SCOPE) or CATALOG_SCOPE
    }
    set_of_optional_params = {AUDIENCE, RESOURCE}
    for param in set_of_optional_params:
        if param_value := self.properties.get(param):
            optional_oauth_param[param] = param_value

    return optional_oauth_param
```